daisi/CMakeLists.txt

```diff
@@ -10,12 +10,19 @@ set(CMAKE_CXX_STANDARD 17)
 cmake_policy(VERSION 3.19)
 
 option(DAISI_DISABLE_NETWORK_SIMULATION "Disable ns-3 network simulation (use ns-3 as discrete event simulator only)" OFF)
+option(DAISI_ENABLE_COMPILER_WARNINGS "Build with compiler warnings. Warnings are handled as errors!" OFF)
 
 # Set default build type to Debug
 if(NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE "Debug")
 endif()
 
+if(DAISI_ENABLE_COMPILER_WARNINGS)
+  add_compile_options(-Wall -Wextra -Wno-sign-compare -Wpedantic -Werror -Wno-maybe-uninitialized)
```

Comment:
Why not enable warnings and Werror separately?

something like this:
```
if(DAISI_ENABLE_COMPILER_WARNINGS OR DAISI_ENABLE_COMPILER_ERRORS)
  add_compile_options(-Wall -Wextra -Wno-sign-compare -Wpedantic -Wno-maybe-uninitialized)
endif()
if(DAISI_ENABLE_COMPILER_ERRORS)
  add_compile_options(-Werror)
endif()
```

Reply:
I added two separate options: one to enable warnings only and one dependent option to enable ``-Werror`` (with CMakeDependentOption).